lib/instrumentation/@elastic/elasticsearch.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2023 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+module.exports = function initialize(_agent, elastic, _moduleName, shim) {
+  shim.setDatastore(shim.ELASTICSEARCH)
+  shim.setParser(queryParser)
+
+  // cwd is node_modules/@elastic/elasticsearch w/ shim.require so use the relative path
+  shim.recordQuery(elastic.Transport.prototype, 'request', function wrapQuery(shim, _, __, args) {
+    const ctx = this
+    return {
+      query: JSON.stringify(args?.[0]),
+      promise: true,
+      opaque: true,
+      inContext: function inContext() {
+        getConnection.call(ctx, shim)
+      }
+    }
+  })
+}
+
+function queryParser(params) {
+  params = JSON.parse(params)
+  const path = params.path.split('/')
+  const method = params.method
```

Comment:
I would think this can be simplified.  The "operation" is the method and then the query is either the querystring or body.  The body probably has to get obfuscated to some degree.